src/compiler/checker.ts

```diff
@@ -14541,9 +14541,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
 
-            const classType = declaration.kind === SyntaxKind.Constructor ?
-                getDeclaredTypeOfClassOrInterface(getMergedSymbol((declaration.parent as ClassDeclaration).symbol))
-                : undefined;
+            const constructorDeclaration = tryGetConstructorDeclaration(declaration);
+            const classType = constructorDeclaration ? getDeclaredTypeOfClassOrInterface(getMergedSymbol((constructorDeclaration.parent as ClassDeclaration).symbol)) : undefined;

```

Comment:
this line is pretty long. Can you return to the original formatting?

Reply:
Reverted